<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>799</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Traffic</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonFollowStyle</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="SimulationView" name="simulationView"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QSlider" name="zoomSlider">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>-10</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="tickPosition">
            <enum>QSlider::NoTicks</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="zoomOutButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../res/icons.qrc">
             <normaloff>:/icons/zoomOut</normaloff>:/icons/zoomOut</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="zoomInButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../res/icons.qrc">
             <normaloff>:/icons/zoomIn</normaloff>:/icons/zoomIn</iconset>
           </property>
           <property name="arrowType">
            <enum>Qt::NoArrow</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="AgentTreeView" name="agentTree">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1071</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoadMap"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Simulation</string>
    </property>
    <addaction name="actionSimSettings"/>
    <addaction name="actionSimStart"/>
    <addaction name="actionSimStop"/>
    <addaction name="actionSimPause"/>
    <addaction name="separator"/>
    <addaction name="actionStatistics"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuVehicle">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Vehicle</string>
    </property>
    <addaction name="actionVehicleRemove"/>
   </widget>
   <widget class="QMenu" name="menuRoad">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Road</string>
    </property>
    <addaction name="actionRoadClose"/>
    <addaction name="actionRoadSetSpeedLimit"/>
    <addaction name="actionRoadSpawnVehicle"/>
   </widget>
   <widget class="QMenu" name="menuJunction">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Junction</string>
    </property>
    <addaction name="actionJunctionClose"/>
    <addaction name="actionJunctionSwitchLights"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionToggleAgentProperties"/>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSimulation"/>
   <addaction name="menuVehicle"/>
   <addaction name="menuRoad"/>
   <addaction name="menuJunction"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoadMap"/>
   <addaction name="actionSimSettings"/>
   <addaction name="separator"/>
   <addaction name="actionSimStart"/>
   <addaction name="actionSimPause"/>
   <addaction name="actionSimStop"/>
   <addaction name="separator"/>
   <addaction name="actionVehicleRemove"/>
   <addaction name="actionRoadClose"/>
   <addaction name="actionRoadSetSpeedLimit"/>
   <addaction name="actionRoadSpawnVehicle"/>
   <addaction name="actionJunctionClose"/>
   <addaction name="actionJunctionSwitchLights"/>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/quit</normaloff>:/icons/quit</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSimStart">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/play</normaloff>:/icons/play</iconset>
   </property>
   <property name="text">
    <string>&amp;Start</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSimPause">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/pause</normaloff>:/icons/pause</iconset>
   </property>
   <property name="text">
    <string>&amp;Pause</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionStatistics">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/statistics</normaloff>:/icons/statistics</iconset>
   </property>
   <property name="text">
    <string>Statistics</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionSimStop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/stop</normaloff>:/icons/stop</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionVehicleRemove">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/remove</normaloff>:/icons/remove</iconset>
   </property>
   <property name="text">
    <string>Remove</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionRoadClose">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/close</normaloff>:/icons/close</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionRoadSetSpeedLimit">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/speedLimit</normaloff>:/icons/speedLimit</iconset>
   </property>
   <property name="text">
    <string>Speed Limit</string>
   </property>
   <property name="toolTip">
    <string>Speed Limit</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionJunctionClose">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/close</normaloff>:/icons/close</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionJunctionSwitchLights">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/lights</normaloff>:/icons/lights</iconset>
   </property>
   <property name="text">
    <string>Switch Lights</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionToggleAgentProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Agent Properties</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionToggleAgentList">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Agent List</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionRoadSpawnVehicle">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/add</normaloff>:/icons/add</iconset>
   </property>
   <property name="text">
    <string>Spawn Vehicle</string>
   </property>
   <property name="toolTip">
    <string>Spawn Vehicle</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSimSettings">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/settings2</normaloff>:/icons/settings2</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionLoadMap">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/world</normaloff>:/icons/world</iconset>
   </property>
   <property name="text">
    <string>&amp;Load Map...</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/zoomIn</normaloff>:/icons/zoomIn</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="toolTip">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>+</string>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="icon">
    <iconset resource="../res/icons.qrc">
     <normaloff>:/icons/zoomOut</normaloff>:/icons/zoomOut</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="toolTip">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SimulationView</class>
   <extends>QGraphicsView</extends>
   <header>SimulationView.h</header>
   <slots>
    <signal>zoomChanged(int)</signal>
    <signal>agentSelected(QString)</signal>
    <slot>zoomIn()</slot>
    <slot>zoomOut()</slot>
    <slot>setZoom(int)</slot>
    <slot>selectAgent(QString)</slot>
    <slot>centerOnAgent(QString)</slot>
    <slot>closeSelected()</slot>
    <slot>removeSelected()</slot>
    <slot>spawnVehicleOnSelected(int)</slot>
    <slot>setSpeedLimitOnSelected(int)</slot>
    <slot>switchLightsOnSelected()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>AgentTreeView</class>
   <extends>QTreeView</extends>
   <header>AgentTreeView.h</header>
   <slots>
    <signal>agentSelected(QString)</signal>
    <slot>selectAgent(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/icons.qrc"/>
  <include location="../res/gfx.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionSimStart</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>startSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>459</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimPause</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>pauseSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>459</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimStop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>459</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomInButton</sender>
   <signal>clicked()</signal>
   <receiver>actionZoomIn</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>533</x>
     <y>762</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomOutButton</sender>
   <signal>clicked()</signal>
   <receiver>actionZoomOut</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>762</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomIn</sender>
   <signal>triggered()</signal>
   <receiver>simulationView</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomOut</sender>
   <signal>triggered()</signal>
   <receiver>simulationView</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>simulationView</receiver>
   <slot>setZoom(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>762</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadMap</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulationView</sender>
   <signal>zoomChanged(int)</signal>
   <receiver>zoomSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>763</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleAgentProperties</sender>
   <signal>toggled(bool)</signal>
   <receiver>agentTree</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>893</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>agentTree</sender>
   <signal>agentSelected(QString)</signal>
   <receiver>simulationView</receiver>
   <slot>selectAgent(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>893</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulationView</sender>
   <signal>agentSelected(QString)</signal>
   <receiver>agentTree</receiver>
   <slot>selectAgent(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>893</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>agentTree</sender>
   <signal>agentSelected(QString)</signal>
   <receiver>simulationView</receiver>
   <slot>centerOnAgent(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>893</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>simulationStarted()</signal>
  <signal>simulationPaused()</signal>
  <signal>simulationStopped()</signal>
  <signal>mapLoaded()</signal>
  <slot>startSimulation()</slot>
  <slot>pauseSimulation()</slot>
  <slot>stopSimulation()</slot>
  <slot>loadMap()</slot>
 </slots>
</ui>
